uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -607,25 +594,39 @@ define([
             return self.directiveOptions.forStaff && loadManagees.call(self);
           })
           .then(function () {
-            if (self.directiveOptions.forStaff && !self.contact) {
-              //throw will break out of chain
-              throw "You have to select a staff to apply absence on their behalf.";
-            }
-
             return loadAbsencePeriods.call(self);
           })
           .then(function () {
             initAbsencePeriod.call(self);
-            self._setMinMaxDate.call(self);
+            self._setMinMaxDate();
 
             return $q.all([
-              self._loadAbsenceTypes.call(self),
               loadCommentsAndContactNames.call(self),
-              self.request.loadAttachments(),
-              self._loadCalendar.call(self)
+              self.request.loadAttachments()
             ]);
           })
           .then(function () {
+            if (self.directiveOptions.forStaff && !self.request.contact_id) {
```

Comment:
why do you need to check the `forStaff` bit? Shouldn't you call the method if there is `contact_id`, period?